packages/core/src/molecule/codec.ts

```diff
@@ -444,8 +445,22 @@ export function union<T extends Record<string, CodecLike<any, any>>>(
   fields?: Record<keyof T, number | undefined | null>,
 ): Codec<UnionEncodable<T>, UnionDecoded<T>> {
   const keys = Object.keys(codecLayout);
+  const values = Object.values(codecLayout);
+  let byteLength = values[0]?.byteLength;
+  for (const { byteLength: l } of values.slice(1)) {
```

Comment:
If the first byteLength is undefined, we can return early to avoid extra checks. Consider to remove the `silce(1)` here.